extensions/standard-processors/tests/unit/SplitJsonTests.cpp

```diff
@@ -0,0 +1,150 @@
+      CHECK_FALSE(flow_file->getAttribute(core::SpecialFlowAttribute::FILENAME).value() == "unique_file_name");
+    }
+  }
+
+  SingleProcessorTestController controller_;
+  core::Processor* split_json_processor_;
+};
+
+TEST_CASE_METHOD(SplitJsonTestFixture, "Query fails with parsing issues", "[SplitJsonTests]") {
+  ProcessorTriggerResult result;
+  std::string error_log;
+  REQUIRE(controller_.plan->setProperty(split_json_processor_, processors::SplitJson::JsonPathExpression, "invalid json path"));
+  SECTION("Flow file content is empty") {
+    result = controller_.trigger({{.content = ""}});
+    error_log = "FlowFile content is empty, transferring to Failure relationship";
+  }
+
+  SECTION("Flow file content is invalid json") {
+    result = controller_.trigger({{.content = "invalid json"}});
+    error_log = "FlowFile content is not a valid JSON document, transferring to Failure relationship";
+  }
+
+  SECTION("Json Path expression is invalid") {
+    result = controller_.trigger({{.content = "{}"}});
+    error_log = "Invalid JSON path expression 'invalid json path' set in the 'JsonPath Expression' property:";
+  }
+
+  CHECK(result.at(processors::SplitJson::Original).empty());
+  CHECK(result.at(processors::SplitJson::Split).empty());
+  CHECK(result.at(processors::SplitJson::Failure).size() == 1);
+  CHECK(utils::verifyLogLinePresenceInPollTime(1s, error_log));
+}
+
+TEST_CASE_METHOD(SplitJsonTestFixture, "Query does not match input JSON content", "[SplitJsonTests]") {
+  REQUIRE(controller_.plan->setProperty(split_json_processor_, processors::SplitJson::JsonPathExpression, "$.email"));
+
+  std::string input_json;
+  SECTION("Flow file content does not contain the specified path") {
+    input_json = R"({"name": "John"})";
+  }
+
+  SECTION("Flow file content null") {
+    input_json = "null";
+  }
+
+  auto result = controller_.trigger({{.content = input_json}});
+
+  CHECK(result.at(processors::SplitJson::Original).empty());
+  CHECK(result.at(processors::SplitJson::Split).empty());
+  CHECK(result.at(processors::SplitJson::Failure).size() == 1);
+  CHECK(utils::verifyLogLinePresenceInPollTime(1s, "JSON Path expression '$.email' did not match the input flow file content, transferring to Failure relationship"));
+}
+
+TEST_CASE_METHOD(SplitJsonTestFixture, "Query returns non-array result", "[SplitJsonTests]") {
+  REQUIRE(controller_.plan->setProperty(split_json_processor_, processors::SplitJson::JsonPathExpression, "$.name"));
+
+  auto result = controller_.trigger({{.content = R"({"name": "John"})"}});
+
+  CHECK(result.at(processors::SplitJson::Original).empty());
+  CHECK(result.at(processors::SplitJson::Split).empty());
+  CHECK(result.at(processors::SplitJson::Failure).size() == 1);
+  CHECK(utils::verifyLogLinePresenceInPollTime(1s, "JSON Path expression '$.name' did not return an array, transferring to Failure relationship"));
```

Comment:
```suggestion
  CHECK(utils::verifyLogLinePresenceInPollTime(1s, "JSON Path expression '$.name' did not return an array, transferring to the 'failure' relationship"));
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1980/commits/0aa867f69a964c3d3f0a1562750bd81218fc6434